go/vt/tabletserver/gorpcqueryservice/sqlquery.go

```diff
@@ -23,13 +23,40 @@ type SqlQuery struct {
 // GetSessionId is exposing tabletserver.SqlQuery.GetSessionId
 func (sq *SqlQuery) GetSessionId(sessionParams *proto.SessionParams, sessionInfo *proto.SessionInfo) (err error) {
 	defer sq.server.HandlePanic(&err)
-	return sq.server.GetSessionId(sessionParams, sessionInfo)
+	tErr := sq.server.GetSessionId(sessionParams, sessionInfo)
+	tabletserver.AddTabletErrorToSessionInfo(tErr, sessionInfo)
+	if *tabletserver.RPCErrorOnlyInReply {
```

Comment:
I'd still prefer to keep the flag, because it's safer/quicker/more easily revertible than just deleting the obsolete code. Once the flag is flipped everywhere, we can go in and delete the unneeded  code (and the flag, eventually).